src/libraries/System.Runtime.InteropServices/gen/LibraryImportGenerator/Analyzers/CustomTypeMarshallerAnalyzer.cs

```diff
@@ -0,0 +1,909 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Immutable;
+using System.Linq;
+using System.Runtime.InteropServices;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+
+using static Microsoft.Interop.Analyzers.AnalyzerDiagnostics;
+
+namespace Microsoft.Interop.Analyzers
+{
+    [DiagnosticAnalyzer(LanguageNames.CSharp, LanguageNames.VisualBasic)]
```

Comment:
I think we want VB support here as otherwise someone could write an invalid marshaller type in VB that would later cause issues for consumption from C#, especially since the shapes of the marshallers don't necessarily require C#-only features (especially if the caller-allocated buffer feature isn't used).

Reply:
I agree. This will only trigger if someone uses one of the attributes right?

Reply:
Yes, this will only trigger if someone uses one of the attribute types.

Reply:
Ah, right.

By 'trigger', you mean actually flag anything, right? The analyzer would still run, but not flag anything.

Reply:
Yeah I mean it won't flag anything.